internal/devconfig/configfile/ast.go

```diff
@@ -425,3 +425,25 @@ func (c *configAST) beforeComment(path ...any) []byte {
 		}),
 	)
 }
+
+func (c *configAST) setEnv(env map[string]string) {
+	members := make([]hujson.ObjectMember, 0, len(env))
+	for k, v := range env {
+		members = append(members, hujson.ObjectMember{
+			Name:  hujson.Value{Value: hujson.String(k)},
+			Value: hujson.Value{Value: hujson.String(v)},
+		})
+	}
+	i := c.memberIndex(c.root.Value.(*hujson.Object), "env")
+	if i == -1 {
+		c.root.Value.(*hujson.Object).Members = append(c.root.Value.(*hujson.Object).Members, hujson.ObjectMember{
+			Name:  hujson.Value{Value: hujson.String("env")},
+			Value: hujson.Value{Value: &hujson.Object{Members: members}},
+		})
+	} else {
+		c.root.Value.(*hujson.Object).Members[i].Value.Value = &hujson.Object{
+			Members: members,
+		}
+	}
+	c.root.Format()
```

Comment:
The above logic seems so convoluted - what's `hujson.ObjectMember`?

Is there a simpler representation? Something like
```
env := [key, val, key, val]
members := make(map[string]string)

for i := 0; i < len(env); i += 2 {
    members[env[i]] = env[i+1]
}
```

Reply:
I'm not sure if we can get rid of any of the code because hujson api is pretty generic so everything needs to be explicit. Maybe  I can refactor some portions out:


```golang
func (c *configAST) createMemberIfMissing(key string) *hujson.ObjectMember {
	i := c.memberIndex(c.root.Value.(*hujson.Object), key)
	if i == -1 {
		c.root.Value.(*hujson.Object).Members = append(c.root.Value.(*hujson.Object).Members, hujson.ObjectMember{
			Name: hujson.Value{Value: hujson.String(key)},
		})
		i = len(c.root.Value.(*hujson.Object).Members) - 1
	}
	return &c.root.Value.(*hujson.Object).Members[i]
}

func mapToObjectMembers(env map[string]string) []hujson.ObjectMember {
	members := make([]hujson.ObjectMember, 0, len(env))
	for k, v := range env {
		members = append(members, hujson.ObjectMember{
			Name:  hujson.Value{Value: hujson.String(k)},
			Value: hujson.Value{Value: hujson.String(v)},
		})
	}
	return members
}

func (c *configAST) setEnv(env map[string]string) {
	c.createMemberIfMissing("env").Value.Value = &hujson.Object{
		Members: mapToObjectMembers(env),
	}
	c.root.Format()
}
```